tests/Extension/CoreTest.php

```diff
@@ -306,6 +306,12 @@ public function provideCompareCases()
             [-1, 42, 'abc42'],
             [-1, 0, 'abc42'],
 
+            [0, 42.0, '   42.0'],
+            [0, 42.0, '42.0   '],
+            [-1, 42.0, '42.0abc'],
+            [-1, 42.0, 'abc42.0'],
+            [-1, 0.0, 'abc42.0'],
```

Comment:
PHP 8 returns 0

Reply:
I copied the existing tests above and only added `.0`s. And this is not the only difference from the current dev state of PHP 8: https://3v4l.org/id8pK/rfc#git-php-master.
But https://github.com/php/php-src/pull/3886 isn't merged (yet), so that's expected, no?